workerserver/finetune.py

```diff
@@ -0,0 +1,114 @@
+import logging
+
+from pydantic import BaseModel
+
+from modelserver.types.remoteworker import FineTuneMethod
+
+
+logger = logging.getLogger(__name__)
+
+
+class FineTuneExecutionPlan(BaseModel):
+    """
+    An execution plan based on the "logical plan" defined by FineTuneJoIn.
+    """
+
+    pytorch_hf_model: str
+    hf_token: str
+    method: FineTuneMethod
+
+    # fsspec-compatible path/URL to the input dataset.
+    # Dataset must be readable using HuggingFace datasets library,
+    # and must have a field where the conversations are stored in "messages" field.
+    dataset_path: str
```

Comment:
currently doesn't support fsspec compat and instead just expects to read from file on disk.

Will add S3 support in a flup